From 941635d60cbc5ff65b91e33df9b6caf6fa47506a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:28 -0400
Subject: [tools] perf/test: Consider PERF_SAMPLE_TRANSACTION in the "sample parsing" test

Message-id: <1408462094-14194-220-git-send-email-jolsa@redhat.com>
Patchwork-id: 88195
O-Subject: [PATCH RHEL7.1 BZ1131394 219/325] perf test: Consider PERF_SAMPLE_TRANSACTION in the "sample parsing" test
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 4ac2f1c1014a121f1493a9d5207258793c576438
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Oct 22 15:28:11 2013 -0300

description
===========
[root@sandy ~]# perf test -v 22
22: Test sample parsing                                    :
--- start ---
sample format has changed, some new PERF_SAMPLE_ bit was introduced - test needs updating
---- end ----
Test sample parsing: FAILED!
[root@sandy ~]#
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/sample-parsing.c b/tools/perf/tests/sample-parsing.c
index 17d000c..61c9da2 100644
--- a/tools/perf/tests/sample-parsing.c
+++ b/tools/perf/tests/sample-parsing.c
@@ -275,7 +275,7 @@ int test__sample_parsing(void)
   * Fail the test if it has not been updated when new sample format bits
   * were added.
   */
- if (PERF_SAMPLE_MAX > PERF_SAMPLE_IDENTIFIER << 1) {
+ if (PERF_SAMPLE_MAX > PERF_SAMPLE_TRANSACTION << 1) {
   pr_debug("sample format has changed, some new PERF_SAMPLE_ bit was introduced - test needs updating\n");
   return -1;
  }
-- 
1.7.1